packages/navigator/src/browser/navigator-filter.ts

```diff
@@ -19,29 +19,29 @@ import { Minimatch } from 'minimatch';
 import { MaybePromise } from '@theia/core/lib/common/types';
 import { Event, Emitter } from '@theia/core/lib/common/event';
 import { PreferenceChangeEvent } from '@theia/core/lib/browser/preferences';
-import { FileNavigatorPreferences, FileNavigatorConfiguration } from './navigator-preferences';
+import { FileSystemPreferences, FileSystemConfiguration } from '@theia/filesystem/lib/browser/filesystem-preferences';
+
+const FILES_EXCLUDE_PREFERENCE: keyof FileSystemConfiguration = 'files.exclude';
 
 /**
  * Filter for omitting elements from the navigator. For more details on the exclusion patterns,
  * one should check either the manual with `man 5 gitignore` or just [here](https://git-scm.com/docs/gitignore).
  */
 @injectable()
 export class FileNavigatorFilter {
-
     protected readonly emitter: Emitter<void> = new Emitter<void>();
 
     protected filterPredicate: FileNavigatorFilter.Predicate;
-
     protected showHiddenFiles: boolean;
 
     constructor(
-        @inject(FileNavigatorPreferences) protected readonly preferences: FileNavigatorPreferences
+        @inject(FileSystemPreferences) protected readonly filesPreferences: FileSystemPreferences
```

Comment:
You should leave the constructor as it was
```js
@inject(FileNavigatorPreferences) protected readonly preferences: FileNavigatorPreferences
```

and use property injection for the `FileSystemPreferences`. Otherwise, you break another API.

Reply:
@kittaakos Thank you for your review. Can you explain more which API it can break?
I don't see any reason to inject `FileNavigatorPreferences` here it is never used in this class

Reply:
> Can you explain more which API it can break?

Sure. Subclasses must call super in the ctors. If there is any subclass of the `FileNavigatorFilter`, you will break client code if we publish a new `latest` version. If clear, please adjust , if no, or I have overlooked something, let me know.

Reply:
@kittaakos Thank you for the explanation, it is clear! I've fixed that.